### DISCUSSION TITLE
set_pcxxheader有办法对uic取消吗

### DISCUSSION THREAD

**Discussion by Elite-stay** [xmake-io/xmake#4852] [Q&A]
示例工程 [xmake_pcxxheader_uic_test.zip](https://github.com/xmake-io/xmake/files/14647411/xmake_pcxxheader_uic_test.zip)

报错是`xmake_pcxxheader_uic_test\src\mainwindow.h(3): fatal error C1083: 无法打开包括文件: “ui_mainwindow.h”: No such file or directory`

推测应该是因为uic需要先生成预编译头，而预编译头又需要uic生成的东西，如果是用vs的qt插件的话，`<QtUic>`的东西默认是不使用预编译头的，这样uic就会先生成，就没有问题

**Comment by star-hengxing** [comment]
xmake 不知道要先 codegen 头文件再编译 pch，但现在有依赖关系就报错了

暂时没有解决方案，可以参考一下这个看有没有思路
https://github.com/xmake-io/xmake/issues/4842

**Comment by waruqi** [comment]
目前不支持，可以在 pch 里面去掉 ui header
